internal/deb/extract.go

```diff
@@ -246,11 +246,22 @@ func extractData(dataReader io.Reader, options *ExtractOptions) error {
 				}
 			}
 			// Create the entry itself.
+			link := tarHeader.Linkname
+			if tarHeader.Typeflag == tar.TypeLink {
+				// The hard link requires the real path of the target file.
+				// For a hard link, we assume:
+				// - the target file is already created (the target file should
+				//   exist before the hard link in a valid tarball)
+				// - the hard link is identified in `fsutil.CreateOptions` as a
+				//   regular file but with a non-empty `Link` field.
```

Comment:
This comment should be in the interface itself (`fsutil.Create`) albeit reworded a bit. We should create the documentation in the function not in the usage in this case.

Reply:
Fixed.